client/message_plaintext.py

```diff
@@ -0,0 +1,68 @@
+import king_phisher.client.plugins as plugins
+import king_phisher.client.gui_utilities as gui_utilities
+
+try:
+	from bs4 import BeautifulSoup
+except ImportError:
+	has_bs4 = False
+else:
+	has_bs4 = True
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Mike Stringer']
+	classifiers = ['Plugin :: Client :: Email :: Message Plaintext']
+	title = 'Message Plaintext'
+	description = """
+	Parse and include a plaintext version of an email based on the HTML version.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = []
+	req_min_version = '1.10.0'
+	version = '1.0'
+	req_packages = {
+		'bs4' : has_bs4
+	}
+	def initialize(self):
+		mailer_tab = self.application.main_tabs['mailer']
+		self.signal_connect('message-create', self.signal_message_create, gobject=mailer_tab)
+		if not has_bs4:
+			return
+
+		if 'message_padding' in self.application.config['plugins']:
+			proceed = gui_utilities.show_dialog_yes_no(
```

Comment:
So what happens when this plugin is enabled **after** the `message_padding` plugin? I'm pretty sure you'll encounter the condition you're trying to avoid. Seems like we need a better way to deal with plugin conflicts. CC @wolfthefallen do you have thoughts here? We could do one of the following that come to mind:

* Add another field to the requirements to mark plugins as incompatible.
* Move this to being a [send-precheck](https://king-phisher.readthedocs.io/en/latest/client/gobject_signals.html#gobject-signal-send-precheck) condition to ensure that the plugin is not enabled just before messages are being sent.
* Add a similar check to the `message_padding` plugin.

What are your thoughts?

Reply:
As this *conflict* is does not prevent either plugin from doing their job, I don't consider it a true conflict. It is more of an opsec violation for phishing that may unintentional make the message look like spam. 

My recommendations:
* Add notes in the description of both plugins stating that use of this plugin with another plugin that changes the plain text part of the message may cause unintentional results. 

Optional:
* As you are the author of the two plugins that may have undesired effects, you can move to send-precheck and do a yes no window asking if you want to continue to use the plugin as it detected another known plugin that could potentially cause undesired effects.

To me plugins are plugins and I would prefer to take the hands off approach to allow the authors to maximize of what they want to do with them, and leave it up to the users of which ones they enable/disable. As long as the plugin does not cause a stack trace or decrease the security of the King Phisher application, I tend not to care about the potential conflict of similar plugins being enabled.

Reply:
I'm in agreement with popping up a dialogue warning for the user to let them know that the two plugins may inadvertently undo what the other is trying to accomplish. It doesn't inherently break anything but...

1. If message padding is used, then the obnoxiously long, random text is exposed in the plaintext version of the email because it depends on HTML to conceal it
2. The plaintext message cannot be different by excluding the HTML-concealed padding or it causes the spam score to increase because the two mime-types are different.

Essentially, one defeats the purpose of the other. I'm in favour of letting the user decide whether to use one or the other or both at their own risk, and just warn them of the consequences. All this _should_ require is one extra change to the message_padding plugin to detect when they are both enabled.

Reply:
Sounds good! FWIW I wouldn't get hung up with concerns about the padding being revealed in the text version for the following reasons:

1) If the message does show up as text, the user won't be able to click the link since there would be no hyperlinks. This would leave them with the option of copying and pasting the URL which I doubt anyone would do thus causing the phish to be ineffective regardless of the trialing padding text.
2) Folks that do have messages being rendered as text are likely security conscious and not going to fall for the message anyways. I have no data to back this up, but it's pretty rare that I see someone disabling HTML rendering for their email and the last time I saw it recommended was to address the GPG vulnerabilities that came out a little bit ago.